### DISCUSSION TITLE
Help with keymaps not working

### DISCUSSION THREAD

**Discussion by A-Lamia** [houmain/keymapper#164] [Q&A]
So currently i have NumLock acting as Virtual3 and Win so when i short press it, it will activate virtual3 and when i long press it, it will be Win "Meta".

The problem I'm having is specific keymaps are not working with NumLock when long pressed for Win but the key is working, but if i actually press the real Windows key followed by my assigned key for my hotkey it works fine ?

```
Win = Meta
Macro = Virtual3
Macro         >> $(msg * /TIME:1 "Macro set")
NumLock{80ms} >> Win
NumLock       >> Macro^

[modifier="Macro"]
Numpad0                  >> $(wt)

# Volume Control: These keys work when using the real Meta Key but not NumLock{80ms} >> Win
Win{NumpadSubtract}      >> AudioVolumeMute^
Win{NumpadDivide}        >> AudioVolumeDown
Win{NumpadMultiply}      >> AudioVolumeUp

# Media Control
NumpadSubtract           >> MediaPlayPause^
NumpadDivide             >> MediaTrackPrevious^
NumpadMultiply           >> MediaTrackNext^


```

any help will be great thank you!

**Comment by A-Lamia** [comment]
Found an other way around this problem.

**Reply by ristomatti** [reply]
Kindly note that it's common courtesy to mention what the workaround was if expecting help from others. Imagine opening a thread with the hope of finding a solution to an issue only to know the OP found a solution....

**Reply by A-Lamia** [reply]
My bad,
I didn't really post anything because it was less a fix for what i was trying to do and more an alternate method.

The "fix" for anyone looking at this post is that I moved to using Numpad0 instead of NumLock (I couldn't get it to work.) as the modifier key, and moved to using a Virtualkey and not Win or Control eg:

```
Macro = Virtual0
MacroLayer = Virtual1
NumLock{200ms} >> NumLock^
NumLock             >> Virtual0^

[modifier="Macro"]
Numpad0                              >> MacroLayer

MacroLayer{NumpadSubtact} >> AudioVolumeMute^
NumpadSubtract                    >> MediaPlayPause^
```

Hope this helps some one :)
